models/discordactions.js

```diff
@@ -114,32 +116,56 @@ const updateDiscordImageForVerification = async (userDiscordId) => {
   }
 };
 
-const getNumberOfMemberForGroups = async (groups = []) => {
+/**
+ * Enriches group data with membership information for a given Discord ID.
+ *
+ * @param {string} discordId - The Discord ID of the user.
+ * @param {Array<object>} groups - Array of group objects to process.
+ * @returns {Promise<Array<object>>} - An array of group objects with enriched information.
+ */
+const enrichGroupDataWithMembershipInfo = async (discordId, groups = []) => {
   try {
     if (!groups.length) {
       return [];
     }
-    const roleIds = groups.map((group) => group.roleid);
 
+    const groupCreatorIds = groups.reduce((ids, group) => {
+      if (group.createdBy) {
+        ids.add(group.createdBy);
+      }
+      return ids;
+    }, new Set());
```

Comment:
Nit: As I understand 
groups look like below
```
 "groups": [
    {
      "id": 
      "date": {
        "_seconds": 
        "_nanoseconds": 
      },
      "rolename": 
      "roleid": 
      "createdBy": 
      "count": 1,
      "firstName": 
      "lastName": 
      "isMember":
      "image": 
    }
  ]
  ```
  
  then ids is in your code is set  in which you are adding createby
  then end result is groupCreatorIds = [createby, createby, , ,...] and it is set
  and created by is the userId right??
  
  Please correct my understanding

Reply:
Yes. I am collecting all the group Creator Ids. But since there can be multiple groups created by the same user, putting in a set gives me a list of unique ids.